examples/chroot-batch

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+set -e
+argv0=chroot-batch
+startdir=$PWD
+OVERWRITE=${OVERWRITE:-0}
+
+# assumption: pkgbase == dirname
+build_args=(-a "$1" -d "$2" --root "$3")
+build_user=${4:-$SUDO_USER}
+db_path=$(realpath -e -- "$db_root/$db_name.db")
+
+# run user/unpriviliged commands with setpriv(1)
+# environment variables are preserved (e.g. LANG=C runas ...)
+runas() { 
+	setpriv --reuid "$1" --regid "$1" --init-groups -- "${@:2}"
+}
+
+# PKGEXT in container may differ from host
+chroot_makepkg_conf=$(aur chroot --path)/etc/makepkg.conf
+build_args+=(--config "$chroot_makepkg_conf")
+
+# 1. check which packages in the queue need to be built
+mapfile -t build_dry_run < <(
+	runas "$build_user" aur build "${build_args[@]}" --dry-run --pkgver)
+wait "$!"
+
+if (( ! OVERWRITE )); then
+    printf '%s\n' "${build_dry_run[@]}" | awk '/^exist:/ {print}'
+
+    mapfile -t pkgbase_targets < <(
+        printf '%s\n' "${build_dry_run[@]}" | awk -F: '/^build:/ {print $2}')
+else
+    mapfile -t pkgbase_targets < <(
+        printf '%s\n' "${build_dry_run[@]}" | awk -F: '{print $2}'
+fi
+
+# 2. chroot preparation
+aur chroot --create --update
+
+batch_sign() {
+	local gpg_args=(--batch --pinentry-mode loopback --passphrase-fd 0)
+	printf '%s' "$2" | runas "$1" command gpg "${gpg_args[@]}" "${@:2}"
+}
+
+repo_add() {
+	LANG=C runas "$1" command repo-add "$2" "${@:3}"
+}
+
+repo_add_and_sign() {
+	# ensure the passphrase is not stored as an environment variable
+	# by saving it in a local variable inside a function
+	local passphrase pkglist pkgbase
+	read -p 'GPG passphrase for signing packages: ' passphrase
+
+	#global var_tmp build_user db_path
+	for pkgbase in "$@"; do
+		cd "$startdir"
+		cd "$pkgbase"
+		PKGDEST="$var_tmp" aur chroot --build --cargs -c,-u # makechrootpkg -cu
+
+		cd "$var_tmp"
+		pkglist=(*) siglist=()
```

Comment:
May need `nullglob` here.

Reply:
I'm trying to think of a scenario where `pkglist` would be empty, but generally yes.